supervisor/dbus/rauc.py

```diff
@@ -69,36 +70,24 @@ def boot_slot(self) -> str | None:
         return self._boot_slot
 
     @dbus_connected
-    def install(self, raucb_file):
-        """Install rauc bundle file.
-
-        Return a coroutine.
-        """
-        return self.dbus.Installer.Install(str(raucb_file))
+    async def install(self, raucb_file) -> None:
+        """Install rauc bundle file."""
+        await self.dbus.Installer.Install(str(raucb_file))
 
     @dbus_connected
-    def get_slot_status(self):
-        """Get slot status.
-
-        Return a coroutine.
-        """
-        return self.dbus.Installer.GetSlotStatus()
+    async def get_slot_status(self) -> list[tuple[str, dict[str, Any]]]:
+        """Get slot status."""
+        return (await self.dbus.Installer.GetSlotStatus())[0]
 
     @dbus_connected
-    def signal_completed(self):
-        """Return a signal wrapper for completed signal.
-
-        Return a coroutine.
-        """
-        return self.dbus.wait_signal(DBUS_SIGNAL_RAUC_INSTALLER_COMPLETED)
+    def signal_completed(self) -> DBusSignalWrapper:
+        """Return a signal wrapper for completed signal."""
+        return self.dbus.signal(DBUS_SIGNAL_RAUC_INSTALLER_COMPLETED)
```

Comment:
The way we were doing this wasn't really safe. It exposed us to a race condition where the signal for `install_completed` fired before we started listening to it. I actually removed the `DBus.wait_signal` method entirely because it seems like any time we need to wait for a response by signal we should be doing it this way instead:
https://github.com/home-assistant/supervisor/blob/72292b01a24e0b362c11bd3f63aceb9d4eb2b56d/supervisor/os/manager.py#L198-L203

It also doesn't work well for continuously listening to signals since the monitor is unattached after every signal. I plan to put in a better approach for that in an upcoming PR.